### ISSUE TITLE
Zabbix 7.0.2 Configuration and CSV have Error Access Denied

### ISSUE THREAD

**Issue by icchoc** [gr8b/zabbix-module-sqlexplorer#13]
Zabbix 7.0.2 used module version 1.12 have Error Access Denied Configuration and CSV (but can use Preview).

![image](https://github.com/user-attachments/assets/e93281bd-97c4-4005-9e6a-f423b12d6ead)

and when saved SQL, don't show name.
![image](https://github.com/user-attachments/assets/fbcff5ae-8d1d-4e32-927d-72e2e8fbe78d)

**Comment by gr8b** [comment]
Cannot reproduce on Zabbix `tags/7.0.2` and module `1.12` version.
Please ensure that content of installed module file `actions/BaseAction.php` in lines range 54-67 match content from [this](https://github.com/gr8b/zabbix-module-sqlexplorer/blob/main/actions/BaseAction.php#L54) file.

**Comment by icchoc** [comment]
Yes, File actions/BaseAction.php in lines range 54-67 match content already and I try replace file again still  the same error.


> Cannot reproduce on Zabbix `tags/7.0.2` and module `1.12` version. Please ensure that content of installed module file `actions/BaseAction.php` in lines range 54-67 match content from [this](https://github.com/gr8b/zabbix-module-sqlexplorer/blob/main/actions/BaseAction.php#L54) file.

**Comment by gr8b** [comment]
Can you please provide more information on:
- php version, webserver (apache, nginx, any other)
- how you installed Zabbix (from git sources, from package)
- do you use module "[Safe mode](https://github.com/gr8b/zabbix-module-sqlexplorer?tab=readme-ov-file#safe-mode)" option

**Comment by icchoc** [comment]
> Can you please provide more information on:
> 
> * php version, webserver (apache, nginx, any other)
- php version 8.3.6
- webserver is apache

> * how you installed Zabbix (from git sources, from package)
- install from git sources

> * do you use module "[Safe mode](https://github.com/gr8b/zabbix-module-sqlexplorer?tab=readme-ov-file#safe-mode)" option
yes, I use Safe mode but not work.

**Comment by gr8b** [comment]
Which one did you used to install Zabbix: tags/7.0.2 or release/7.0

**Comment by icchoc** [comment]
> Which one did you used to install Zabbix: tags/7.0.2 or release/7.0

Zabbix 7.0.2

**Comment by Godd67** [comment]
I have same issue on 6.4.17, installed in docker with version 1.10 . Any hints how to workaround?
Also, side question - what is a proper way to upgrade to a new module version? Copy over/wipe first etc? Is it backward compatible ?

**Comment by gr8b** [comment]
To install a different version of the module, you need to delete the directory with the old version and add the new version. Then, click "Scan directory" on the page with the list of modules.
Yes, module is backward compatible, queries created with one version will be available in other version of module.

**Comment by gr8b** [comment]
> I have same issue on 6.4.17, installed in docker with version 1.10 . Any hints how to workaround?

I would suggest to install latest version of module because the issue you reported were fixed in version 1.11.

**Comment by gr8b** [comment]
I tried to reproduce the issue on version 6.4.17 using the latest version of the module (1.12) but could not. Here is what I did:

```sh
git worktree add ../tags/6.4.17 tags/6.4.17
cd ../tags/6.4.17
# create 6.4.17 database
cd ui/modules
curl -OL https://github.com/gr8b/zabbix-module-sqlexplorer/releases/download/v1.12/v1.12-zabbix-6.4-7.0.zip
mkdir zabbix-module-frontent-sqlexplorer
unzip v1.12-zabbix-6.4-7.0.zip -d zabbix-module-frontent-sqlexplorer/
```

**Comment by Godd67** [comment]
I upgraded to 1.12 , but it did not help... I guess it is my setup - I run Zabbix in docker - could it be user permission issue?

**Comment by Godd67** [comment]
Actually, now behavior is different - I am still getting "Admin " permission errors, but if I run SQL , i'm getting Cloudflare block :(
Have you seen this before ? Any workarounds?
We run zabbix via CF DNS record.
![image](https://github.com/user-attachments/assets/be94b040-70a4-4bd1-b3b8-166e64a089d6)

**Comment by gr8b** [comment]
> I upgraded to 1.12 , but it did not help... I guess it is my setup - I run Zabbix in docker - could it be user permission issue?

This sounds strange because issue is in how module determine Zabbix version it is run on. Do you have custom dockerfile? Is it possible to get minimal version, to be able to reproduce?

**Comment by gr8b** [comment]
> Actually, now behavior is different - I am still getting "Admin " permission errors, but if I run SQL , i'm getting Cloudflare block :( Have you seen this before ? Any workarounds? We run zabbix via CF DNS record. ![image](https://private-user-images.githubusercontent.com/1334713/361241863-be94b040-70a4-4bd1-b3b8-166e64a089d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2ODAyMjUsIm5iZiI6MTcyNDY3OTkyNSwicGF0aCI6Ii8xMzM0NzEzLzM2MTI0MTg2My1iZTk0YjA0MC03MGE0LTRiZDEtYjNiOC0xNjZlNjRhMDg5ZDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjZUMTM0NTI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTMyMWZlMmM0MGMwNmIyNzQ3MGM4ZmI5MjRkZTg3MTE2OTlmZjE1N2U0NjQ1ZmVjNmJhZjEyMmMxYWE0YTc0MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.4KICVEr1IuVs4INr6PTUECeDfvDTb3wiAujKm9D64cA)

CF is blocking request with sql similar content? 😮 
If it is so i am afraid there is nothing could be done from module side because it should somehow transfer query to PHP side to execute.. Only.. I cat try to make a workaround if you would help with testing it.

**Comment by Godd67** [comment]
It may be a custom rule for company's DNS, Yes, I can test it if it does not involve any complex stuff.

**Comment by gr8b** [comment]
Can you test 1.12 from attached .zip?
[v1.12-zabbix-6.4-7.0.zip](https://github.com/user-attachments/files/16752831/v1.12-zabbix-6.4-7.0.zip)

**Comment by gr8b** [comment]
I am still interested in reproducing the reported error. Could you help by providing the following information?
- Constants `ZABBIX_VERSION`, `ZABBIX_API_VERSION` and `ZABBIX_EXPORT_VERSION `defined in `defines.inc.php` file

**Comment by Godd67** [comment]
Sorry , took me long time - I still have not used new 1.12 but it starts working for me at least partially and it is fully working for my colleague . 
- CF stopped blocking  SQL
- Configuration does not work for me , but works for my coworker, our users seems similar - both have SA role and System default Frontend Access.

The only difference I see is that he works from companies VPN.

**Comment by Godd67** [comment]
> I am still interested in reproducing the reported error. Could you help by providing the following information?
> 
>     * Constants `ZABBIX_VERSION`, `ZABBIX_API_VERSION` and `ZABBIX_EXPORT_VERSION `defined in `defines.inc.php` file

define('ZABBIX_VERSION',                '6.4.17');
define('ZABBIX_API_VERSION',    '6.4.17');
define('ZABBIX_EXPORT_VERSION', '6.4');

**Comment by gr8b** [comment]
> Sorry , took me long time - I still have not used new 1.12 but it starts working for me at least partially and it is fully working for my colleague .
> 
> * CF stopped blocking  SQL
> * Configuration does not work for me , but works for my coworker, our users seems similar - both have SA role and System default Frontend Access.
> 
> The only difference I see is that he works from companies VPN.

Module do not have limitation based on user type and/or role. Can you provide what is submitted to server from browser when you receive "Access denied" error?

**Comment by gr8b** [comment]
@Godd67 do you have any news on issue?
